panther_gpiod/src/gpio_driver.cpp

```diff
@@ -0,0 +1,274 @@
+// Copyright 2023 Husarion sp. z o.o.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "panther_gpiod/gpio_driver.hpp"
+
+#include <mutex>
+#include <string>
+#include <thread>
+#include <vector>
+
+#include <poll.h>
+
+#include <realtime_tools/thread_priority.hpp>
+
+namespace panther_gpiod
+{
+
+GPIODriver::GPIODriver(std::vector<GPIOInfo> gpio_info, int gpio_monit_thread_sched_priority)
+: gpio_info_storage_(std::move(gpio_info)),
+  gpio_monit_thread_sched_priority_(gpio_monit_thread_sched_priority)
+{
+  if (gpio_info_storage_.empty()) {
+    throw std::runtime_error("Empty GPIO info vector provided");
+  }
+
+  auto gpio_chip = gpiod::chip(gpio_chip_path_);
+  line_request_ = create_line_request(gpio_chip);
+
+  gpio_monitor_on();
```

Comment:
It seems to me that this might be inconvenient for the user. When monitoring is turned off, the 'is_pin_active' method will not provide accurate information. Do we want to add a check in it to see if the thread is enabled? If so, what should it return then?